### PR TITLE
WIP: Attempt to migrate from Travis CI to GitHub Actions

### LINKED ISSUE #72: Fix the Appveyor build
https://ci.appveyor.com/project/ndmitchell/filepath is failing. The problem is that it doesn't find `time` on the plan. Trying the same project but renaming `filepath` to anything else fixes the issue, so it seems to be that by putting `filepath` in the plan it invalidates `time` somehow. A Stack bug, but I'm not convinced one they'll necessarily want to fix, as overwriting `filepath` is an odd thing to do.

### PR DESCRIPTION
**Pull request by Rufflewind**
This is an attempt to migrate from Travis CI to GitHub Actions, since travis-ci.org is being shut down. Unfortunately I wasn't able to figure out how to make the neil tool work yet. Would appreciate any guidance on that.

Logs: https://github.com/Rufflewind/filepath/actions/workflows/ci.yml

- [x] I had to bump to the latest Cabal version because the `neil` GitHub Action uses flags that don't exist in Cabal 2.4.
- [x] Bumped QuickCheck to allow 2.14 and thereby GHC 9.0. ~~It complains that GHC 9.0 must be included in tested-with, but I don't think <s>Criterion</s> QuickCheck 2.13 is compatible with GHC 9.0 yet?~~
- [ ] Eventually it fails with an error about not being able to find the `Generate` module in `travis.hs` (copied to `.neil/src/Main.hs`):

~~~
+ cabal v2-install --allow-newer --flags=small --installdir=. --install-method=copy --overwrite-policy=always

...

src/Main.hs:2:1: error:
Error:     Could not find module ‘Generate’
    Use -v to see a list of the files searched for.
  |
2 | import Generate
  | ^^^^^^^^^^^^^^^
Error: Process completed with exit code 1.
~~~

This PR would resolve #72, if it worked.

### CODE REVIEW

**Comment by RyanGlScott** [comment]
> It complains that GHC 9.0 must be included in tested-with, but I don't think Criterion is compatible with GHC 9.0 yet?

`criterion` fully supports GHC 9.0, so that shouldn't be an issue.

**Comment by Rufflewind** [comment]
Sorry, I meant to say `QuickCheck >= 2.7 && < 2.14`, which prevents filepath tests from building on GHC 9.0.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          - { os: ubuntu-latest, ghc: 8.6.5 }
+          - { os: ubuntu-latest, ghc: 8.8.4 }
+          - { os: ubuntu-latest, ghc: 8.10.2 }
+          - { os: ubuntu-18.04, ghc: head }
+          - { os: windows-latest, ghc: 9.0 }
+    runs-on: ${{ matrix.os }}
+    steps:
+      - if: startsWith(matrix.os, 'ubuntu-')
+        run: |
+          sudo add-apt-repository ppa:hvr/ghc
```
Comment (hasufell) [inline .github/workflows/ci.yml:27]: I don't think this is needed anymore?

**Comment by hasufell** [comment]
Fixed in https://github.com/haskell/filepath/pull/89

**Pull request closed without merging**